languages/swift/generic/Parse_swift_tree_sitter.ml

```diff
@@ -2448,32 +2448,36 @@ and map_primary_expression (env : env) (x : CST.primary_expression) : G.expr =
       let v1 = (* "\\" *) token env v1 in
       let v2 =
         match v2 with
-        | Some x -> (
-            match x with
-            | `Simple_user_type x ->
-                let id, targs = map_simple_user_type env x in
-                let name = H2.name_of_id id in
-                let name = H2.add_type_args_opt_to_name name targs in
-                G.TyN name |> G.t
-            | `Array_type x -> map_array_type env x
-            | `Dict_type x -> map_dictionary_type env x)
-        | None -> todo env ()
+        | Some x ->
+            G.T
+              (match x with
+              | `Simple_user_type x ->
+                  let id, targs = map_simple_user_type env x in
+                  let name = H2.name_of_id id in
+                  let name = H2.add_type_args_opt_to_name name targs in
+                  G.TyN name |> G.t
+              | `Array_type x -> map_array_type env x
+              | `Dict_type x -> map_dictionary_type env x)
+        | None ->
+            G.T (G.OtherType (("Implicit", Tok.unsafe_fake_tok ""), []) |> G.t)
```

Comment:
This is not good, because $X will not have any real token in it, which will make fail certain things.
Maybe you should reuse v1 (the token for \\) at least.
Try running this rule in semgrep instead of semgrep-core, and use --json; I think you would get an error
because semgrep will not be able to generate a content for $X

Reply:
Thanks for pointing this out. Replaced the unsafe fake token with `v1`.